dht/dht.go

```diff
@@ -208,8 +238,8 @@ func (nid *nodeID) ByteString() string {
 	return string(buf[:])
 }
 
-type node struct {
-	addr          dHTAddr
+type Node struct {
```

Comment:
Hm, the inconsistency is because I exposed Node but neglected (yet) to expose its attributes. My reasoning again is from use-case; passive monitoring nodes, or active search nodes, etcetera. So, `Node.Id` (currently `Node.id`) is something I'd like hooks to have access to, as well as `Node.Addr`. I'd even expose `Node.AnnounceToken` for informational purposes, but not the `last..` bits.

The reasoning is, there are no good 'research tool' libraries for DHT analysis, so even if I can't think why someone might want to know about a particular datum, unless it's fairly important for it to remain private I feel like making it public would make `torrent` more useful and distinctive. I initially tried to make changes of this kind to `github.com/nictuku/dht` but would have had to refactor the entire thing messily to add those features. The modular nature of `torrent` makes it much easier to add DHT-only features without breaking the rest of it.

Without making Node public, I could find other ways to expose information though; for example, by attaching methods to Msg that allow it to fetch Msg-requester information, or by creating a special struct for "QueryHandler" information and populating it only when a hook is found from private information known to the server, then passing it in. That might allow one to expose information useful to hook users without fixing a `Node` public interface that might impose a maintenance burden, if that's your concern?